url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/Selecting_Algothrim/lecMSMomentum.java

```diff
@@ -0,0 +1,384 @@
+package Selecting_Algothrim;
+
+import java.util.LinkedList;
+import java.lang.Math;
+
+import Trading_Engine.MyAskList;
+import Trading_Engine.MyBidList;
+import Trading_Engine.ResultData;
+
+import java.sql.Time;
+
+public class lecMSMomentum implements lecMS {
+
+	public lecMSMomentum(){
+		counting = 0;
+		arrayIsFull = false;
+		mode = BUY_MODE;
+		average = 0;
+		shareQuantity = 0;
+		lastAverage = 0;
+	}
+	
+	
+	
+	
+	public void setThreShold( double number){
+		this.epsilon = number;
+	}
+	
+	public void setAmountToTrade(int number){
+		this.shareQuantityToTrade = number;
+	}
+	
+	
+	public void addTrade( double trade){
+		//once Counting reach 9 , wihch is SIZE_OF_ARRAY, we know arrayIsFull
+		if (( arrayIsFull == false) && counting == SIZE_OF_ARRAY) { // just to check if should it take average or not
+			arrayIsFull = true;
+		}
+		// this makes counting equals to zero, so it works like a round array
+		if ( counting == SIZE_OF_ARRAY ) { // to see if counting reachs 9
+			counting = 0;
+		}
+		// insert the trade inside the tenArray
+		tenArray[counting] = trade;
+		counting++;
+		// take Average only when arrayIsFull
+		if ( arrayIsFull){
+			lastAverage = average;
+			average = 0;
+			for ( int i = 0 ; i < SIZE_OF_ARRAY ; i++ ){
+				//System.out.println(i + " value: " + tenArray[i]);
+				average += tenArray[i]; 
+			}
+			average = average/10;
+			averageChange = (average - lastAverage)/ lastAverage;
+		}
+	}
+	public LinkedList<signalObject> generateSignalList( LinkedList<orderObject> buyOrders, LinkedList<orderObject> sellOrders){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.